### PR TITLE
Add Student model

### PR DESCRIPTION
**Pull request by Ansel-Ch**
I implemented the `Student` model with reference to the `Person` model. There were several modifications to the instance fields to make it relevant to students.

### CODE REVIEW

**Comment by Ansel-Ch** [comment]
The new implementation inherits from the `Person` class. So hopefully, it is substitutable with the Person instances in the code. Do let me know what yall think.

File: src/main/java/seedu/address/model/student/Student.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        this.studentNumber = studentNumber;
+    }
+
+    // public Name getName() {
+    //     return name;
+    // }
+    //
+    // public ContactNumber getContactNumber() {
+    //     return contactNumber;
+    // }
```
Comment (Edsel-Tan) [inline src/main/java/seedu/address/model/student/Student.java:46]: Probably can remove this?

File: src/test/java/seedu/address/model/student/ContactNumberTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        // null phone number
+        assertThrows(NullPointerException.class, () -> ContactNumber.isValidContact(null));
+
+        // invalid phone numbers
+        assertFalse(ContactNumber.isValidContact("")); // empty string
+        assertFalse(ContactNumber.isValidContact(" ")); // spaces only
+        assertFalse(ContactNumber.isValidContact("91")); // less than 3 numbers
+        assertFalse(ContactNumber.isValidContact("phone")); // non-numeric
+        assertFalse(ContactNumber.isValidContact("9011p041")); // alphabets within digits
+        assertFalse(ContactNumber.isValidContact("9312 1534")); // spaces within digits
```
Comment (Edsel-Tan) [inline src/test/java/seedu/address/model/student/ContactNumberTest.java:33]: Might be abit too strict?

**Review by Edsel-Tan — COMMENTED** [review]
Should discuss if we allow phone numbers before merging

File: src/main/java/seedu/address/model/student/Student.java (hunk 1 above)
Comment (Ansel-Ch) [reply src/main/java/seedu/address/model/student/Student.java:46]: Sure

File: src/test/java/seedu/address/model/student/ContactNumberTest.java (hunk 2 above)
Comment (Ansel-Ch) [reply src/test/java/seedu/address/model/student/ContactNumberTest.java:33]: Yea it probably is. However, ContactNumber is currently not being used so we could probably make regex changes if we decide to use it in our Student model in the future.

Comment (Ansel-Ch) [reply src/test/java/seedu/address/model/student/ContactNumberTest.java:33]: I can just comment it out for now and we can decide on this when we decide to use the contactNumber.

**Comment by gnoossk** [comment]
Im merging it.

**Pull request merged**